### PR TITLE
fix(traces): remove nan from log_evaluations

### LINKED ISSUE #2054: [BUG] log_evaluations usability issue with nans
If you log scores containing nan values, those scores will be accepted without error by `log_evaluations`, but Phoenix will break when loading the UI. We should either:
- fail the call to `log_evaluations` when a user attempts to log nans with an informative error message
- accept nans to `log_evaluations`, but filter them out so that the subsequent graphql query does not break
I'd vote for the latter option.

Note: This also seems to happen with explanations, although nans are less likely to occur in that scenario.

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves #2054

### CODE REVIEW

File: src/phoenix/session/evaluation.py [hunk 1]
```diff
@@ -103,8 +103,14 @@ def _extract_subject_id_from_index(
...
     score = cast(Optional[float], row.get("score"))
+    if isinstance(score, float) and math.isnan(score):
+        score = None
     label = cast(Optional[str], row.get("label"))
+    if isinstance(label, float) and math.isnan(label):
+        label = None
     explanation = cast(Optional[str], row.get("explanation"))
+    if isinstance(explanation, float) and math.isnan(explanation):
+        explanation = None
     if (
```
Comment (axiomofjoy) [inline src/phoenix/session/evaluation.py:114]: There is some redundancy here.

**Review by axiomofjoy — APPROVED** [review]

File: src/phoenix/session/evaluation.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/session/evaluation.py:114]: (on line 115)

Comment (RogerHYang) [reply src/phoenix/session/evaluation.py:114]: which part is redundant?

Comment (RogerHYang) [reply src/phoenix/session/evaluation.py:114]: ok i see. will remove

**Pull request merged**